### ISSUE TITLE
Enable Export Option within Editor

### ISSUE THREAD

**Issue by MackenzieHartung** [ampproject/amp-wp#2464]
As a user I want to see a CTA to allow me to export my AMP Story from within the editor.

AC1: The CTA should appear within the right panel of the AMP Stories editor
AC2: A user should not see the ability to export (perhaps the option is disabled) until the AMP Story is "saved".

**Comment by amedina** [comment]
The export/download option can be available contextually based on user roles. For example, `admin` users can be in charge of exporting stories to be loaded on different platforms; they would have the responsibility of guide/tailor the download to their target destination.

**Comment by westonruter** [comment]
@amedina Do you think the export button would be limited by role or the _enabling_ of the export feature would be limited by role? In other words, assuming there needs to be some configuration to provide the base URL for where the story will be exported to. If this configuration can only be supplied by an admin, e.g. when this option is selected:

![image](https://user-images.githubusercontent.com/134745/58728112-0cd58480-839b-11e9-97aa-18d320c97bd5.png)

Then I think the export button could be displayed for any user role, no?

**Comment by amedina** [comment]
Great point. I agree. That matches the use case of story creators on given org can create and export stories which can then be uploaded to a structure configured (when exporting is enabled) by the admin.

**Comment by dawidmlynarz** [comment]
From a design perspective, I would add another action button (next to Publish) with an export icon that could trigger Exporting/Download action. See attachment below for that:

<img width="557" alt="Screenshot 2019-06-03 at 08 51 37" src="https://user-images.githubusercontent.com/8026823/58781707-ed2d9e80-85dc-11e9-879a-bab7e422a3d7.png">

I kept it simple and secondary as we don't want to do too much mess around the primary call to action which is Publish.

Let me know what you think!

**Comment by swissspidy** [comment]
Looking at https://github.com/WordPress/gutenberg/blob/392bf02030ff4b25c478972911ca51ea21eef975/docs/designers-developers/developers/slotfills/plugin-sidebar.md we might not be able to _easily_ add a button there without also having a custom sidebar for it (see screenshot on that page). Just noting that here.

**Comment by westonruter** [comment]
A possible UI for the enabling story exporting:

![image](https://user-images.githubusercontent.com/134745/58824539-240daf80-85f1-11e9-8d4c-6e0bed6870ad.png)

Two additional considerations that I think need to be made here:

* Links to other stories may also need to be rewritten. If there is a bookend (#2442) or CTA grid layer that links to other stories, they should get the same base URL so that the links don't break. This is going to be needed at least for the `rel=canonical` link at least.
* It is possible that the base URL for assets should be different than the base URL for non-assets. Consider a site that wants to put all media files into a common directory (like WordPress does) rather than storing them in a subdirectory under the URL route for the story. _This also may very well be out of scope for this initial feature._ But I wanted to note it.

**Comment by valendesigns** [comment]
I suggest we use the built-in tools and options sidebar instead. I've added the Architectural Summary to #2466 which has a proposed UI, including what you've added above @westonruter. However,  it uses the [PluginMoreMenuItem](https://github.com/WordPress/gutenberg/blob/master/docs/designers-developers/developers/slotfills/plugin-more-menu-item.md) instead of the [PluginSidebar](https://github.com/WordPress/gutenberg/blob/master/docs/designers-developers/developers/slotfills/plugin-sidebar.md), which has its own panel. I moved it because if we put the icon in the header then we have to add a panel that we don't need. If we want to move it later we can, but this will help speed up the first version and keep things a bit simpler.
